build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/dependencies/patches/Utils.java

```diff
@@ -101,9 +109,15 @@ public static void patchJar(File inputFile, File outputFile, Collection<PatcherI
                         );
                     }
                 } else {
-                    // Read the entry's data and write it to the new JAR
-                    try (InputStream is = jarFile.getInputStream(entry)) {
-                        is.transferTo(jos);
+                    if (unsignJar && entryName.equals("META-INF/MANIFEST.MF")) {
+                        var manifest = new Manifest(jarFile.getInputStream(entry));
+                        manifest.getEntries().clear();
+                        manifest.write(jos);
+                    } else if (unsignJar == false || entryName.matches("META-INF/.*\\.SF") == false) {
```

Comment:
the Jar I'm patching includes a `MSTFSIG.SF`, which I assume will be named differently in other signed JARs

Reply:
Umm if you need to perform this kind of manipulation, maybe it's even better to have a more generic
`BiFunction<Boolean, InputStream, OutputStream>`

The boolean will tell if you handled the entry or not:
```
try (InputStream is = jarFile.getInputStream(entry)) {
   if (function.apply(is, jos) == false) {
      // by default, copy
      is.transferTo(jos);
   }
}
```

Reply:
Also, if you go down that route (you need that), I will consider refactoring this so that the transformers fit into this too... IDK if this is something you want to do though.

Reply:
yeah, I think this path can get very complicated; it's probably better to wait until someone needs to transform a Manifest file in a different way (or another META-INF file) and this can be restructured to support that